hersh/reader.test.cpp

```diff
@@ -0,0 +1,29 @@
+#include <hersh/reader.hpp>
+
+#include <iostream>
+#include <ostream>
+#include <sstream>
+
+#include <catch2/catch_test_macros.hpp>
+
+TEST_CASE("Test Reader", "[Reader]") {
+    std::istringstream istringstream("alon");
+    std::ostringstream ostringstream;
+
+    hersh::Reader reader(istringstream, ostringstream);
+
+    REQUIRE(reader.getPrompt() == "");
```

Comment:
I'm not sure this is a behavior worth testing... Is it important to us that the default prompt is an empty string?

Reply:
I think it's important to test that the class does what you think it does. If you think the default prompt is empty, you should check that. That's how I see it.

Reply:
That means that every time we change the default prompt we need to change the test. I think this test causes more problems than it solves.

Reply:
Yes, you'd have to change the test, because you change the class. I don't see a problem with that. 

In any case, I don't think we'll change the default prompt a lot. Maybe one more time to make it more consistent with bash ( just '$' instead of 'hersh$').

Reply:
When is this test useful? When you make a typo in the default prompt? It causes trouble and will never catch any bugs.

Reply:
Done.